### PR TITLE
Sjs/fix typescript coverage

### LINKED ISSUE #2374: launcher/cli: coveralls isn't mapping back to TypeScript
For example: https://coveralls.io/builds/11707972/source?filename=chrome-launcher%2Fchrome-finder.js

Because the coverage was gathered in JS. Need to remap to get type lines of TS accounted for.
https://www.sitepen.com/blog/2015/09/29/code-coverage-for-typescript-and-other-transpiled-languages/

### PR DESCRIPTION
**Pull request by samccone**
use nyc + lcov merge to generate a combined coverage report.

Fixes #2374

### CODE REVIEW

**Comment by samccone** [comment]
neato working!
https://coveralls.io/builds/12023306/source?filename=chrome-launcher%2Fchrome-launcher.ts

File: chrome-launcher/package.json [hunk 1]
```diff
@@ -4,22 +4,25 @@
   "scripts": {
     "build": "tsc",
     "dev": "tsc -w",
-    "test": "mocha --require ts-node/register --reporter dot test/**/*-test.ts",
+    "test": "nyc mocha --require ts-node/register --reporter=dot test/**/*-test.ts --timeout=10000",
```
Comment (patrickhulce) [inline chrome-launcher/package.json]: I generally like to leave test as the regular command which makes coverage just `nyc npm test` or potentially `nyc npm test && nyc report`

Comment (samccone) [inline chrome-launcher/package.json]: done

**Comment by paulirish** [comment]
needs a rebase after the logger commit landed.

but more importantly, this is conflicting with #2519 a bit.
im gonna try to resolve the two, but i might pull you in.

**Comment by paulirish** [comment]
Manually landed cc5485feadc5bd228ddedc43b5d572e6b309bdf0 since the branch is conflicted with the logger.

**Pull request closed without merging**